src/libpmemkv.h

```diff
@@ -30,132 +30,53 @@
-
-    virtual void AllBetween(void* context, const std::string& key1, const std::string& key2, KVAllCallback* callback) = 0;
-    void AllBetween(const std::string& key1, const std::string& key2, std::function<KVAllFunction> f);
-    void AllBetween(const std::string& key1, const std::string& key2, std::function<KVAllStringFunction> f);
-
-    virtual int64_t Count() = 0;
-    virtual int64_t CountAbove(const std::string& key) = 0;
-    virtual int64_t CountBelow(const std::string& key) = 0;
-    virtual int64_t CountBetween(const std::string& key1, const std::string& key2) = 0;
-
-    virtual void Each(void* context, KVEachCallback* callback) = 0;
-    void Each(std::function<KVEachFunction> f);
-    void Each(std::function<KVEachStringFunction> f);
-
-    virtual void EachAbove(void* context, const std::string& key, KVEachCallback* callback) = 0;
-    void EachAbove(const std::string& key, std::function<KVEachFunction> f);
-    void EachAbove(const std::string& key, std::function<KVEachStringFunction> f);
-
-    virtual void EachBelow(void* context, const std::string& key, KVEachCallback* callback) = 0;
-    void EachBelow(const std::string& key, std::function<KVEachFunction> f);
-    void EachBelow(const std::string& key, std::function<KVEachStringFunction> f);
-
-    virtual void EachBetween(void* context, const std::string& key1, const std::string& key2, KVEachCallback* callback) = 0;
-    void EachBetween(const std::string& key1, const std::string& key2, std::function<KVEachFunction> f);
-    void EachBetween(const std::string& key1, const std::string& key2, std::function<KVEachStringFunction> f);
-
-    virtual KVStatus Exists(const std::string& key) = 0;
-
-    virtual void Get(void* context, const std::string& key, KVGetCallback* callback) = 0;
-    void Get(const std::string& key, std::function<KVGetFunction> f);
-    void Get(const std::string& key, std::function<KVGetStringFunction> f);
-    KVStatus Get(const std::string& key, std::string* value);
-
-    virtual KVStatus Put(const std::string& key, const std::string& value) = 0;
-    virtual KVStatus Remove(const std::string& key) = 0;
-};
-
 extern "C" {
 #endif
 
 #include <stdint.h>
 #include <stddef.h>
 
-struct KVEngine;
-typedef struct KVEngine KVEngine;
+#include "status.h"
 
-KVEngine* kvengine_start(void* context, const char* engine, const char* config, KVStartFailureCallback* callback);
-void kvengine_stop(KVEngine* kv);
+typedef struct pmemkv_engine pmemkv_engine;
 
-void kvengine_all(KVEngine* kv, void* context, KVAllCallback* c);
-void kvengine_all_above(KVEngine* kv, void* context, int32_t kb, const char* k, KVAllCallback* c);
-void kvengine_all_below(KVEngine* kv, void* context, int32_t kb, const char* k, KVAllCallback* c);
-void kvengine_all_between(KVEngine* kv, void* context, int32_t kb1, const char* k1,
-                          int32_t kb2, const char* k2, KVAllCallback* c);
+typedef void(pmemkv_all_callback_t)(void* context, int keybytes, const char* key);
```

Comment:
I think () around the name is not needed
If it is please add space after return type.